### PR TITLE
k8s: Avoid printing warning when fast retries of CNP status updates fail

### PR DESCRIPTION
**Pull request by tgraf**
Only warn after the max number of fast retries fail as failures are normal while the resource is under heavy contention.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5756)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

File: daemon/k8s_watcher.go [hunk 1]
```diff
@@ -1809,16 +1811,19 @@ func cnpNodeStatusController(ciliumV2Store cache.Store, cnp *cilium_v2.CiliumNet
...
 
 	if cnpUpdateErr != nil {
-		return cnpUpdateErr
+		overallErr = cnpUpdateErr
 	} else {
-		return waitForEPsErr
+		overallErr = waitForEPsErr
 	}
+
+	logger.WithError(overallErr).Warningf("Update of CNP status failed %s times. Will keep retrying.", maxAttempts)
```
Comment (aanm) [inline daemon/k8s_watcher.go]: Won't this always be printed regardless `overallErr` is or isn't nil? Also, `failed %s times` -> `failed %d times`

**Review by aanm — CHANGES_REQUESTED** [review]

File: daemon/k8s_watcher.go (hunk 1 above)
Comment (tgraf) [reply daemon/k8s_watcher.go]: 🤦‍♀️ Fixed

**Comment by tgraf** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by tgraf** [comment]
Rebased after merge conflict

**Comment by tgraf** [comment]
test-me-please

EDIT: Legit CI failure to what appears to be a regression from ff830ed. Reverted commit and rebased.

**Comment by tgraf** [comment]
test-me-please

**Pull request merged**